File: RecyclerViews/RecyclerViewDemo3/app/src/main/java/edu/cs4730/recyclerviewdemo3/Simple3_Fragment.java
package edu.cs4730.recyclerviewdemo3;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.Arrays;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import edu.cs4730.recyclerviewdemo3.databinding.Simple3FragmentBinding;


/**
 * A third "simple" example of a recycler view.   Everything with this example is prefix simple3_
 * this uses a cardview as the layout, so it looks "better" and you see where each view is separated.
 */
public class Simple3_Fragment extends Fragment {
    Simple3FragmentBinding binding;
    Simple3_myAdapter mAdapter;
    List<String> values = Arrays.asList("Android", "iPhone", "WindowsMobile",
        "Blackberry", "WebOS", "Ubuntu", "Windows7", "Max OS X",
        "Linux", "OS/2");


    public Simple3_Fragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = Simple3FragmentBinding.inflate(inflater, container, false);
        //setup the RecyclerView
        binding.list.setLayoutManager(new LinearLayoutManager(requireContext()));
        binding.list.setItemAnimator(new DefaultItemAnimator());
        //setup the adapter, which is myAdapter, see the code.
        mAdapter = new Simple3_myAdapter(values, requireContext());
        //add the adapter to the recyclerview
        binding.list.setAdapter(mAdapter);

        return binding.getRoot();
    }

}

File: RecyclerViews/RecyclerViewDemo3_kt/app/src/main/java/edu/cs4730/recyclerviewdemo3_kt/Simple3_Fragment.kt
package edu.cs4730.recyclerviewdemo3_kt

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.DefaultItemAnimator
import edu.cs4730.recyclerviewdemo3_kt.databinding.Simple3FragmentBinding
import java.util.*

/**
 * A third "simple" example of a recycler view.   Everything with this example is prefix simple3_
 * this uses a cardview as the layout, so it looks "better" and you see where each view is separated.
 */
class Simple3_Fragment : Fragment() {
    lateinit var binding: Simple3FragmentBinding
    lateinit var mAdapter: Simple3_myAdapter
    var values = Arrays.asList(
        "Android", "iPhone", "WindowsMobile",
        "Blackberry", "WebOS", "Ubuntu", "Windows7", "Max OS X",
        "Linux", "OS/2"
    )

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        binding = Simple3FragmentBinding.inflate(inflater, container, false)
        //setup the RecyclerView
        binding.list.layoutManager = LinearLayoutManager(requireContext())
        binding.list.itemAnimator = DefaultItemAnimator()
        //setup the adapter, which is myAdapter, see the code.
        mAdapter = Simple3_myAdapter(values, R.layout.simple3_rowlayout, requireContext())
        //add the adapter to the recyclerview
        binding.list.adapter = mAdapter
        return binding.root
    }
}